### ISSUE TITLE
InvalidChangeBatch when deleting records for a ingress update from 3.xx to 6.xx

### ISSUE THREAD

**Issue by fbarrerafalabella** [bitnami/charts#11811]
### Name and Version

bitnami/external-dns 6.7.0

### What steps will reproduce the bug?

Deploy an ingress and let external dns create records on 3.2.3 version, update to 6.7.0 try deleting the ingress, wait for external dns to delete the records and it will try to delete 6 when the previous version just created 4, it will result in an error

### What is the expected behavior?

delete the records

### What do you see instead?
```
time="2022-08-17T19:40:38Z" level=info msg="Desired change: DELETE api-skeleton-373.docs.reviewapps.linio-staging.com A [Id: /hostedzone/Z2I7IGA46DMTXB]"
time="2022-08-17T19:40:38Z" level=info msg="Desired change: DELETE api-skeleton-373.reviewapps.linio-staging.com A [Id: /hostedzone/Z2I7IGA46DMTXB]"
time="2022-08-17T19:40:38Z" level=info msg="Desired change: DELETE external-dns.a-api-skeleton-373.docs.reviewapps.linio-staging.com TXT [Id: /hostedzone/Z2I7IGA46DMTXB]"
time="2022-08-17T19:40:38Z" level=info msg="Desired change: DELETE external-dns.a-api-skeleton-373.reviewapps.linio-staging.com TXT [Id: /hostedzone/Z2I7IGA46DMTXB]"
time="2022-08-17T19:40:38Z" level=info msg="Desired change: DELETE external-dns.api-skeleton-373.docs.reviewapps.linio-staging.com TXT [Id: /hostedzone/Z2I7IGA46DMTXB]"
time="2022-08-17T19:40:38Z" level=info msg="Desired change: DELETE external-dns.api-skeleton-373.reviewapps.linio-staging.com TXT [Id: /hostedzone/Z2I7IGA46DMTXB]"
time="2022-08-17T19:40:38Z" level=error msg="Failure in zone linio-staging.com. [Id: /hostedzone/Z2I7IGA46DMTXB]"
time="2022-08-17T19:40:38Z" level=error msg="InvalidChangeBatch: [Tried to delete resource record set [name='external-dns.a-api-skeleton-373.docs.reviewapps.linio-staging.com.', type='TXT'] but it was not found, Tried to delete resource record set [name='external-dns.a-api-skeleton-373.reviewapps.linio-staging.com.', type='TXT'] but it was not found]\n\tstatus code: 400, request id: 9b8a7a60-1cc5-4ffb-b96c-83fe1f1d2bb1"
time="2022-08-17T19:40:38Z" level=error msg="failed to submit all changes for the following zones: [/hostedzone/Z2I7IGA46DMTXB]"
```

**Comment by marcosbc** [comment]
Hi @fbarrerafalabella, please note that between versions 3.x and 6.x of the chart, there have been breaking changes (e.g. in the chart's side or in the external-dns's side). Due to this, I'm afraid that there isn't much we can do on our side to fix the issue.

**Comment by fbarrerafalabella** [comment]
Hey marcos, I created also an issue directly with external-dns seems like this was a documented issue that is solved on 0.12.1 version (I took 6.7.5 from bitnami and the problem was solved) 
https://github.com/kubernetes-sigs/external-dns/issues/2958#issuecomment-1219162778
